api/tests/middleware/authorization-middleware.test.ts

```diff
@@ -0,0 +1,83 @@
+import { Response, NextFunction } from "express"
+import { User } from "@/models"
+import { auth0Integration, type Auth0UserInfo } from "@/integrations/auth0-integration"
+import {
+  authorizationMiddleware,
+  type AuthorizationRequest,
+} from "@/middleware/authorization-middleware"
+
+jest.unmock("@/middleware/authorization-middleware")
+
+jest.mock("@/integrations/auth0-integration")
+jest.mock("@/utils/logger")
+
+let user_data: Auth0UserInfo
+let req: Partial<AuthorizationRequest>
+let res: Partial<Response>
+let next: NextFunction
+
+beforeEach(() => {
+  user_data = {
+    email: "dupe@test.com",
+    firstName: "dupe",
+    lastName: "UNKNOWN",
+    auth0Subject: "",
+  }
+
+  req = {
+    headers: {
+      authorization: "mock-token",
+    },
+    auth: {},
+  }
+
+  res = {
+    status: jest.fn().mockReturnThis(),
+    json: jest.fn(),
+  }
+
+  next = jest.fn()
+})
+
+async function setupRaceCondition(auth0Subject: string, attempts: number) {
+  user_data.auth0Subject = auth0Subject
+  req.auth = { sub: auth0Subject }
+
+  // force a "new" user in the middleware
+  jest.spyOn(auth0Integration, "getUserInfo").mockResolvedValue(user_data)
+
+  // Act
+  const funcs = Array.from({ length: attempts }, async () => {
+    await authorizationMiddleware(req as AuthorizationRequest, res as Response, next)
+  })
+
+  await Promise.all(funcs)
+}
```

Comment:
This is mixin an "Arrange" and "Act".

I'd expect the "Arrange" code to return the list of functions, and the "Act" part to do the `await Promise.all(funcs)`

i.e.

```
// Arrange
user_data.auth0Subject = auth0Subject
  req.auth = { sub: auth0Subject }

  // force a "new" user in the middleware
  jest.spyOn(auth0Integration, "getUserInfo").mockResolvedValue(user_data)

  // Act
  const funcs = Array.from({ length: attempts }, async () => {
    await authorizationMiddleware(req as AuthorizationRequest, res as Response, next)
  })
  
// Act
await Promise.all(funcs)

// Expect
...
```

I think it will add clarity to the test if you add this code inline in the test.